homeassistant/components/jewish_calendar/sensor.py

```diff
@@ -171,40 +238,30 @@ async def async_setup_entry(
     async_add_entities: AddConfigEntryEntitiesCallback,
 ) -> None:
     """Set up the Jewish calendar sensors ."""
-    sensors = [
+    sensors: list[JewishCalendarBaseSensor] = [
         JewishCalendarSensor(config_entry, description) for description in INFO_SENSORS
     ]
     sensors.extend(
         JewishCalendarTimeSensor(config_entry, description)
         for description in TIME_SENSORS
     )
-
     async_add_entities(sensors)
 
 
-class JewishCalendarSensor(JewishCalendarEntity, SensorEntity):
-    """Representation of an Jewish calendar sensor."""
+class JewishCalendarBaseSensor(JewishCalendarEntity, SensorEntity):
+    """Base class for Jewish calendar sensors."""
 
     _attr_entity_category = EntityCategory.DIAGNOSTIC
 
-    def __init__(
-        self,
-        config_entry: JewishCalendarConfigEntry,
-        description: SensorEntityDescription,
-    ) -> None:
-        """Initialize the Jewish calendar sensor."""
-        super().__init__(config_entry, description)
-        self._attrs: dict[str, str] = {}
-
     async def async_added_to_hass(self) -> None:
         """Call when entity is added to hass."""
         await super().async_added_to_hass()
-        await self.async_update()
+        await self.async_update_data()
 
-    async def async_update(self) -> None:
```

Comment:
why was this called `async_update`?

Reply:
Because previously it would actually update the state of the sensor. (See below the assignment to `self._attr_native_value`)
This PR changes the behavior as now it updates the results class which is then read from the native_value property.